Property undefined error in type definitions.

## Steps to Reproduce

Just import `apollo-client`.
### Buggy Behavior

```
[0] /home/zhenghu/coding/qsc/qsc-main-react-typescript/node_modules/apollo-client/index.d.ts
[0] (47,18): error TS2459: Type '{ networkInterface?: NetworkInterface | undefined; reduxRootKey?: string | undefined; reduxRootSe...' has no property 'networkInterface' and no string index signature.
[0] /home/zhenghu/coding/qsc/qsc-main-react-typescript/node_modules/apollo-client/index.d.ts
[0] (47,36): error TS2459: Type '{ networkInterface?: NetworkInterface | undefined; reduxRootKey?: string | undefined; reduxRootSe...' has no property 'reduxRootKey' and no string index signature.
[0] /home/zhenghu/coding/qsc/qsc-main-react-typescript/node_modules/apollo-client/index.d.ts
[0] (47,50): error TS2459: Type '{ networkInterface?: NetworkInterface | undefined; reduxRootKey?: string | undefined; reduxRootSe...' has no property 'reduxRootSelector' and no string index signature.
[0] /home/zhenghu/coding/qsc/qsc-main-react-typescript/node_modules/apollo-client/index.d.ts
[0] (47,69): error TS2459: Type '{ networkInterface?: NetworkInterface | undefined; reduxRootKey?: string | undefined; reduxRootSe...' has no property 'initialState' and no string index signature.
[0] /home/zhenghu/coding/qsc/qsc-main-react-typescript/node_modules/apollo-client/index.d.ts
[0] (47,83): error TS2459: Type '{ networkInterface?: NetworkInterface | undefined; reduxRootKey?: string | undefined; reduxRootSe...' has no property 'dataIdFromObject' and no string index signature.
[0] /home/zhenghu/coding/qsc/qsc-main-react-typescript/node_modules/apollo-client/index.d.ts
[0] (47,101): error TS2459: Type '{ networkInterface?: NetworkInterface | undefined; reduxRootKey?: string | undefined; reduxRootSe...' has no property 'queryTransformer' and no string index signature.
[0] /home/zhenghu/coding/qsc/qsc-main-react-typescript/node_modules/apollo-client/index.d.ts
[0] (47,119): error TS2459: Type '{ networkInterface?: NetworkInterface | undefined; reduxRootKey?: string | undefined; reduxRootSe...' has no property 'resultTransformer' and no string index signature.
[0] /home/zhenghu/coding/qsc/qsc-main-react-typescript/node_modules/apollo-client/index.d.ts
[0] (47,138): error TS2459: Type '{ networkInterface?: NetworkInterface | undefined; reduxRootKey?: string | undefined; reduxRootSe...' has no property 'resultComparator' and no string index signature.
[0] /home/zhenghu/coding/qsc/qsc-main-react-typescript/node_modules/apollo-client/index.d.ts
[0] (47,156): error TS2459: Type '{ networkInterface?: NetworkInterface | undefined; reduxRootKey?: string | undefined; reduxRootSe...' has no property 'shouldBatch' and no string index signature.
[0] /home/zhenghu/coding/qsc/qsc-main-react-typescript/node_modules/apollo-client/index.d.ts
[0] (47,169): error TS2459: Type '{ networkInterface?: NetworkInterface | undefined; reduxRootKey?: string | undefined; reduxRootSe...' has no property 'ssrMode' and no string index signature.
[0] /home/zhenghu/coding/qsc/qsc-main-react-typescript/node_modules/apollo-client/index.d.ts
[0] (47,178): error TS2459: Type '{ networkInterface?: NetworkInterface | undefined; reduxRootKey?: string | undefined; reduxRootSe...' has no property 'ssrForceFetchDelay' and no string index signature.
[0] /home/zhenghu/coding/qsc/qsc-main-react-typescript/node_modules/apollo-client/index.d.ts
[0] (47,198): error TS2459: Type '{ networkInterface?: NetworkInterface | undefined; reduxRootKey?: string | undefined; reduxRootSe...' has no property 'mutationBehaviorReducers' and no string index signature.
[0] /home/zhenghu/coding/qsc/qsc-main-react-typescript/node_modules/apollo-client/index.d.ts
[0] (47,224): error TS2459: Type '{ networkInterface?: NetworkInterface | undefined; reduxRootKey?: string | undefined; reduxRootSe...' has no property 'batchInterval' and no string index signature.
```
### Expected Behavior

No error.
### Version
- apollo-client@0.4.21
- react-apollo@0.5.11

--- a/AUTHORS
+++ b/AUTHORS
@@ -32,6 +32,7 @@ Oleksandr Stubailo <sashko@Oleksandrs-MacBook-Pro.local>
 Olivier Ricordeau <olivier@ricordeau.org>
 Pavol Fulop <pavol.fulop@regex.sk>
 Pavol Fulop <pavolfulop@gmail.com>
+Rasmus Eneman <rasmus@eneman.eu>
 Robin Ricard <ricard.robin@gmail.com>
 Sashko Stubailo <s.stubailo@gmail.com>
 Sashko Stubailo <sashko@stubailo.com>

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,13 +4,14 @@ Expect active development and potentially significant breaking changes in the `0
 
 ### vNEXT
 
+- Fix typings error with `strictNullChecks` [PR #1188](https://github.com/apollostack/apollo-client/pull/1188)
 - ...
 
 ### 0.7.3
 - *Fixed breaking change:* readQueryFromStore was incomptibale with Typescript 2.0 compiler. [PR #1171](https://github.com/apollostack/apollo-client/pull/1171)
 
 ### 0.7.2
-Re-release of 0.7.1 with proper internal directory structure 
+Re-release of 0.7.1 with proper internal directory structure
 
 ### 0.7.1
 - *Undo breaking change:* Add whatwg-fetch polyfill (most likely only until version 1.0) [PR #1155](https://github.com/apollostack/apollo-client/pull/1155)

--- a/src/ApolloClient.ts
+++ b/src/ApolloClient.ts
@@ -132,22 +132,7 @@ export default class ApolloClient {
    * with identical parameters (query, variables, operationName) is already in flight.
    *
    */
-  constructor({
-    networkInterface,
-    reduxRootKey,
-    reduxRootSelector,
-    initialState,
-    dataIdFromObject,
-    resultComparator,
-    ssrMode = false,
-    ssrForceFetchDelay = 0,
-    mutationBehaviorReducers = {} as MutationBehaviorReducerMap,
-    addTypename = true,
-    resultTransformer,
-    customResolvers,
-    connectToDevTools,
-    queryDeduplication = false,
-  }: {
+  constructor(options: {
     networkInterface?: NetworkInterface,
     reduxRootKey?: string,
     reduxRootSelector?: string | ApolloStateSelector,
@@ -163,6 +148,22 @@ export default class ApolloClient {
     connectToDevTools?: boolean,
     queryDeduplication?: boolean,
   } = {}) {
+    let {
+      networkInterface,
+      reduxRootKey,
+      reduxRootSelector,
+      initialState,
+      dataIdFromObject,
+      resultComparator,
+      ssrMode = false,
+      ssrForceFetchDelay = 0,
+      mutationBehaviorReducers = {} as MutationBehaviorReducerMap,
+      addTypename = true,
+      resultTransformer,
+      customResolvers,
+      connectToDevTools,
+      queryDeduplication = false,
+    } = options;
     if (reduxRootKey && reduxRootSelector) {
       throw new Error('Both "reduxRootKey" and "reduxRootSelector" are configured, but only one of two is allowed.');
     }

